src/components/structures/MessagePanel.js

```diff
@@ -115,6 +115,18 @@ module.exports = React.createClass({
         // to manage its animations
         this._readReceiptMap = {};
 
+        // Track read receipts by user ID. For each user ID we've ever shown a
+        // a read receipt for, we store an object:
+        //   {
+        //       lastShownEventId,
```

Comment:
minor nit: for documentation of structures it would be nice to show the type being stored, so something like:

```
{
   lastShownEventId: string,
   receipt: { ...whatever a receipt is... },
}
```

Reply:
Good idea, added.